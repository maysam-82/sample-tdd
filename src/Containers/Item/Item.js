import React, { Component } from 'react';
import { Form } from 'react-bootstrap';

class Item extends Component {
	constructor(props) {
		super(props);

		this.state = {
			person: '',
			itemName: '',
		};
	}
	onInputChanged = event => {
		const {
			target: { value },
		} = event;
		this.setState({ person: event.target.value });
	};

	render() {
		return (
			<div className="container">
				<Form>
					<Form.Group>
						<Form.Label>Person Name</Form.Label>
						<Form.Control className="input-person" onChange={event => this.onInputChanged(event)} />
					</Form.Group>
				</Form>
			</div>
		);
	}
}

export default Item;
