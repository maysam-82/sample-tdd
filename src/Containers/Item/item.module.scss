.itemContainer {
	border: 0.0625rem solid #ccc;
	padding: 0.25rem 1rem;
	margin-bottom: 1rem;
}
