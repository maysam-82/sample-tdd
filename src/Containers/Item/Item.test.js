import React from 'react';
import { shallow } from 'enzyme';
import Item from './Item';

describe('Item Component', () => {
	const item = shallow(<Item />);
	it('should render correctly', () => {
		expect(item.debug()).toMatchSnapshot();
	});
	it('initializes a person and its item in `state`', () => {
		expect(item.state()).toEqual({ person: '', itemName: '' });
	});
	describe('while entering name of person in input', () => {
		const personName = 'Father';
		beforeEach(() => item.find('.input-person').simulate('change', { target: { value: personName } }));
		it('should update `person` in `state`', () => {
			expect(item.state().person).toEqual(personName);
		});
	});
});
