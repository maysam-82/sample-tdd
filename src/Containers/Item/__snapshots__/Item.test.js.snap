// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Item Component should render correctly 1`] = `
"<div className=\\"container\\">
  <Form inline={false}>
    <FormGroup>
      <FormLabel column={false} srOnly={false}>
        Person Name
      </FormLabel>
      <FormControl className=\\"input-person\\" onChange={[Function: onChange]} />
    </FormGroup>
  </Form>
</div>"
`;
